modules/auxiliary/scanner/karaf/login.rb

```diff
@@ -0,0 +1,158 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'net/ssh'
+require 'metasploit/framework/login_scanner/ssh'
+require 'metasploit/framework/credential_collection'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::CommandShell
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name'        => 'Karaf Default Credential Scanner',
+      'Description' => %q{
+        This module uses default Karaf credentials to login to the console via ssh.
+      },
+      'Author'      => [
+          'Samuel Huckins',
+          'Brent Cook',
+          'Peer Aagaard',
+          'Greg Mikeska',
+          'Dev Mohanty'
+      ],
+      'License'     => MSF_LICENSE
+    )
+
+    register_options(
+      [
+        # TODO Set default user, pass
+        Opt::RPORT(8101),
+        OptString.new('USERNAME', [true, 'Username', 'karaf']),
+        OptString.new('PASSWORD', [true, 'Password', 'karaf'])
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        Opt::Proxies,
+        OptBool.new('STOP_ON_SUCCESS', [ false, '', true]),
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+
+  end
+
+  def rport
+    datastore['RPORT']
+  end
+
+  def session_setup(result, ssh_socket)
+    return unless ssh_socket
+
+    # Create a new session
+    conn = Net::SSH::CommandStream.new(ssh_socket, '/bin/sh', true)
+
+    merge_me = {
+      'USERNAME'      => result.credential.public,
+      'PASSWORD'      => result.credential.private
+    }
+    info = "#{proto_from_fullname} #{result.credential} (#{@ip}:#{datastore['RPORT']})"
+    s = start_session(self, info, merge_me, false, conn.lsock)
+
+    # Set the session platform
+    case result.proof
+      when /Linux/
+        s.platform = "linux"
+      when /Darwin/
+        s.platform = "osx"
+      when /SunOS/
+        s.platform = "solaris"
+      when /BSD/
+        s.platform = "bsd"
+      when /HP-UX/
+        s.platform = "hpux"
+      when /AIX/
+        s.platform = "aix"
+      when /Win32|Windows/
+        s.platform = "windows"
+      when /Unknown command or computer name/
+        s.platform = "cisco-ios"
+    end
+
+    s
+  end
+
+  def gather_proof
+    proof = ''
+    begin
+      Timeout.timeout(5) do
+        proof = ssh_socket.exec!("shell:info\n").to_s
+      end
+    rescue ::Exception
+    end
+    proof
+  end
+
+  def run_host(ip)
+    @ip = ip
+    print_status("Attempting login to #{ip}:#{rport}...")
+
+    cred_collection = Metasploit::Framework::CredentialCollection.new(
```

Comment:
@dmohanty-r7 Currently, your module is designed to only check one user/pass, but I think it is more capable than that.

I suggest adding these things:

1. Support all the user/pass options as we normally would (:pass_file, :user_file, :userpass_file, etc)
2. Register a boolean datastore option allowing the user to try the default "karaf" credential
3. If the datastore option is true, then do:

```ruby
if datastore['TRYDEFAULTCRED']
  cred_collection.additional_privates << 'karef'
  cred_collection.additional_publics << 'karef'
end
```